### PR TITLE
Issue #64

### PR DESCRIPTION
**Pull request by maximilian02**
- Implementing auth0 sign up, sign in and logout on the front end with [auth0-angular](https://github.com/auth0/auth0-angular)
- Adding backend auth0 token validation with [express-jwt](https://github.com/auth0/express-jwt)

Issue link: https://github.com/MakingSense/mean-seed/issues/64

### CODE REVIEW

File: test/frontend-unit-tests/spec/controllers/login.js [hunk 1]
```diff
@@ -7,40 +7,72 @@ describe('Controller: LoginCtrl', function () {
...
-  it('should set scope.errorMessage on mongoose errors', function () {
-    $httpBackend.expectPOST('/auth/').respond(404, {
-      message: 'Test Error'
+  it('should initialize controller with variables defined', function () {
+    expect(typeof LoginCtrl).toBe('object');
+    expect(scope.errorMessage).toBeDefined();
+    expect(scope.submitted).toBeDefined();
+    expect(scope.submitted).toBeFalsy();
+    expect(scope.errorMessage).toEqual('');
+  });
```
Comment (mravinale) [inline test/frontend-unit-tests/spec/controllers/login.js]: I like the initialization check :)

File: README.md [hunk 2]
```diff
@@ -18,6 +18,9 @@ npm install
 
 Read more about installing [Node](https://nodejs.org/download/), [Bower](http://bower.io/) and [Grunt client](http://gruntjs.com/getting-started).
 
+Environment variables:
+Remember to set up Auth0 credentials using the ".env" file. Please go ahead and rename the existent ".env.example" file and fill it with the proper information.
+
```
Comment (mravinale) [inline README.md]: good catch :)

File: test/frontend-unit-tests/spec/controllers/login.js [hunk 3]
```diff
@@ -7,40 +7,72 @@ describe('Controller: LoginCtrl', function () {
...
-      username: ''
-    };
+    LoginCtrl = $controller('LoginCtrl', {
+      $scope: scope,
+      $rootScope: rootScope,
+      $location: location,
+      userService: UserService
+    });
+
+    scope.user = {};
```
Comment (mravinale) [inline test/frontend-unit-tests/spec/controllers/login.js]: why this is void?

Comment (maximilian02) [inline test/frontend-unit-tests/spec/controllers/login.js]: It shouldn't be there. Now we handle the validation on the view side, allowing the user to login only when the user and password are complete.

**Comment by maximilian02** [comment]
Finally, no errors!

File: .gitignore [hunk 4]
```diff
@@ -8,4 +8,5 @@ public/dist/
 public/styles/*.css
 public/styles/*.css.map
 coverage
-.settings
\ No newline at end of file
+.settings
+.env
```
Comment (mravinale) [inline .gitignore]: for ignoring this file should use this : 
`*.env`
or
`/.env`

**Comment by mravinale** [comment]
:+1:

**Pull request merged**